### DISCUSSION TITLE
Possible to change depth on transformers?

### DISCUSSION THREAD

**Discussion by Onyoursix** [explosion/spaCy#7313] [Help: Coding & Implementations] [answered]
Is it possible to change the depth when training NER with transformers like you can on the tok2vec model as shown below? 

[components.tok2vec.model.encode]
@architectures = "spacy.MaxoutWindowEncoder.v2"
width = 256
depth = 8
window_size = 1
maxout_pieces = 3

**Comment by honnibal** [comment, marked as answer]
The transformers architecture assumes that the transformer is being by name (or path) via the `transformers.AutoModel` class, which means it's a defined architecture including the number of layers.

If you want to define something more custom, you could make your own PyTorch model and wrap it with Thinc. But most users will want to use a transformer with pretrained weights, which you can't do if you make your own model.

An alternative would be to leave the transformer alone, and inside your components, add a few extra transformer layers, so that the component has some weights of its own that aren't transfer-learnt.

**Reply by Onyoursix** [reply]
Good to know, I'll have to give your alternative idea a shot.

Thanks!
